lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/searcher/RemoteRepositoryIndexSearcher.java

```diff
@@ -113,14 +113,19 @@ public RemoteRepositoryIndexSearcher(MavenLemminxExtension lemminxMavenPlugin, P
 				return indexDownloadJobs.get(res).thenApply(theVoid -> context);
 			}
 			final IndexingContext context = initializeContext(repositoryUrl);
-			indexingContexts.put(repositoryUrl, context);
-			CompletableFuture<IndexingContext> future;
-			try {
-				future = updateIndex(context).thenApply(theVoid -> context);
-				indexDownloadJobs.put(context, future);
-				return future;
-			} catch (ComponentLookupException e) {
-				return CompletableFuture.failedFuture(e);
+			if (context != null) {
```

Comment:
What's actually changed here in the context of the dependency resolution? We should avoid unrelated changes to ease review and improve traceability.

Reply:
Same as previous. Can be removed or moved to another PR, if acceptable